src/lib/updatePollMessage.ts

```diff
@@ -0,0 +1,77 @@
+import {
+    IModify,
+    IPersistence,
+    IRead,
+} from "@rocket.chat/apps-engine/definition/accessors";
+import {
+    RocketChatAssociationModel,
+    RocketChatAssociationRecord,
+} from "@rocket.chat/apps-engine/definition/metadata";
+
+import { createPollBlocks } from "./createPollBlocks";
+import { getPoll } from "./getPoll";
+
+export async function updatePollMessage({
+    data,
+    read,
+    persistence,
+    modify,
+}: {
+    data;
+    read: IRead;
+    persistence: IPersistence;
+    modify: IModify;
+}) {
+    const poll = await getPoll(String(data.view.id), read);
+
+    if (!poll) {
+        throw new Error("no such poll");
+    }
+
+    if (poll.finished) {
+        throw new Error("this poll is already finished");
+    }
+
+    try {
+        const {
+            state,
+        }: {
+            state?: any;
+        } = data.view;
+
+        poll.options.push(state.addOption.option);
```

Comment:
Don't we need a null/empty check before add the new option?

Reply:
Missed this case! Thanks.